misk/src/main/kotlin/misk/client/ClientLoggingInterceptor.kt

```diff
@@ -0,0 +1,47 @@
+package misk.client
+
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
+import wisp.logging.getLogger
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton internal class ClientLoggingInterceptor @Inject constructor(
+  private val httpClientsConfig: HttpClientsConfig
+): Interceptor {
+  override fun intercept(chain: Interceptor.Chain): Response {
+    val result = chain.proceed(chain.request())
+
+    if (httpClientsConfig.logRequests) {
+      val outgoingRequest = result.request
+      logger.info { "Request: ${outgoingRequest.url}, headers=${headers(outgoingRequest)}" }
+    }
+
+    return result
+  }
+
+  private fun headers(request: Request): Map<String, String> =
+    request.headers
+      .filter { (key, _) -> LOGGED_HEADERS.contains(key.lowercase()) }
+      .toMap()
+
+  companion object {
+    val logger = getLogger<ClientLoggingInterceptor>()
+
+    val LOGGED_HEADERS = listOf(
+      "content-type",
+      "user-agent",
+      "content-length",
```

Comment:
Based on the test - these aren't accessible in request.headers? (I'm ok with that, since I'm really just interested in the tracing headers.